docs/adr/001-lambda-scope.md

```diff
@@ -0,0 +1,16 @@
+# 1. Lambda scope
+
+Date: 2023-05-09
+
+## Status
+
+Proposed
```

Comment:
@ehanson8 - FWIW, I'd be comfortable with changing this to "Accepted" with this PR as the proposal + discussion activity.